### PR TITLE
Add new transport, winston-cloudwatch

### PR DESCRIPTION
**Pull request by rplute**
Adds Cloudwatch transport for audits, uncaught exceptions, and errors, which is only activated when environment variable LOG_ENV (i.e.., process.env.LOG_ENV) is set to something other than development. This is so development environments don't push stuff to AWS CloudWatch. Later, when I don't have AWS environments as "development", I'll switch this to NODE_ENV and get rid of LOG_ENV.

Creates log group in AWS Cloudwatch based on LOG_ENV name for the following:

test-api-audit-group
test-api-error-group
test-api-exceptions-group

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,12 +51,13 @@ let initialized = false;
  * }
  */
 function KLog(opt) {
+
   opt = opt || {};
 
   opt.transports = opt.transports || [];
   opt.exceptionHandlers = opt.exceptionHandlers || [];
 
-  this.application = opt.application;
```
Comment (baconinthemorn) [inline index.js]: Assuming we'll need to update usage of this in our other apps...?

File: index.js [hunk 2]
```diff
@@ -104,24 +115,64 @@ function KLog(opt) {
...
+
       if (options.transports[i].type === 'Splunk') {
         transports.push(new(splunk)(options.transports[i].options));
+
       } else if (options.transports[i].type === 'Mail') {
         transports.push(new(winston.transports.Mail)(options.transports[i].options));
+
+      } else if (options.transports[i].type === 'CloudWatch') {
+
+        // Based on process.env.LOG_ENV. Default 'development', which does not send to cloudwatch
```
Comment (baconinthemorn) [inline index.js]: Would love to see some comment(s) here describing what this is accomplishing.

File: index.js [hunk 3]
```diff
@@ -104,24 +115,64 @@ function KLog(opt) {
...
   if (options.exceptionHandlers) {
     for (let i = 0; i < options.exceptionHandlers.length; i++) {
+
       if (options.exceptionHandlers[i].type === 'Mail') {
         exceptionHandlers.push(new(winston.transports.Mail)(options.exceptionHandlers[i].options));
+
+      } else if (options.exceptionHandlers[i].type === 'CloudWatch') {
+
+        // Based on process.env.LOG_ENV. Default 'development', which does not send to cloudwatch
+        if (env !== 'development') {
```
Comment (baconinthemorn) [inline index.js]: You're doing basically the same thing above... perhaps throw it into a function to keep it DRY?

Comment (rplute) [inline index.js]: This part is for Exception Handlers, which is different than the transports section. Different logic altogether.

File: index.js (hunk 2 above)
Comment (rplute) [inline index.js]: Development environment (local machine development) shouldn't send to AWS Cloudwatch. This prevents it.

File: index.js (hunk 1 above)
Comment (rplute) [inline index.js]: As said in Slack, the eConsent Suite had the name of the app as title when passing configs.

File: index.js (hunk 3 above)
Comment (rplute) [inline index.js]: Unless you're talking about the push itself. I can see what you mean, I could potentially put it in a Function.

Comment (baconinthemorn) [inline index.js]: Yeah was speaking to the push itself - not a huge snippet, but potentially could be conserved into a single function. Up to you if you'd like to refactor it into a single function.

File: index.js (hunk 2 above)
Comment (baconinthemorn) [inline index.js]: I meant add comments for the contents of the if statement, like we spoke of on Slack, just a blurb that describes what the crypto module is doing.

File: index.js (hunk 3 above)
Comment (rplute) [inline index.js]: After looking at the code for a little bit, I agree that transports and exceptionhandlers could run through the same function. Not just the push, but the entire for loop. This just is how it was originally written. At a later time, I'd be more than happy to do this.

File: index.js (hunk 2 above)
Comment (rplute) [inline index.js]: Ah, yeah, I can do that.

File: index.js (hunk 3 above)
Comment (baconinthemorn) [inline index.js]: Sounds good!

**Pull request merged**